org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/natives/ExceptionPropagationInNativeTestCase.xtend

```diff
@@ -0,0 +1,39 @@
+package org.uqbar.project.wollok.tests.natives
+
+import org.uqbar.project.wollok.tests.interpreter.AbstractWollokInterpreterTestCase
+import org.junit.Test
+
+/**
+ * @author jfernandes
+ */
+// TODO: this could have more tests, with more complex cases :P
+class ExceptionPropagationInNativeTestCase extends AbstractWollokInterpreterTestCase {
+	
+	@Test
+	def void nativeMethodCallsWollokClosureThrowingASimpleException() {
+		#["natives"->'''
+class Repeater {
+	method ntimes(n, closure) native
+}
+
+program natives {
+	const repeater = new Repeater()
+
+	const closure = { throw new Exception("SoundGarden") }
+	try {
+		repeater.ntimes(1, closure)
+	}
+	catch e {
+		assert.equals(
+"wollok.lang.Exception: SoundGarden
+	at  [natives.wpgm]
+	at natives.Repeater.ntimes(n,closure) [natives.wpgm]
+	at  [natives.wpgm]
+",e.getStackTraceAsString())
+	} 
+}
+		''']
```

Comment:
Yeah the problem is that if I had tabs rich strings will take them as part
of the string.
I know that sometimes it recognize a piece of rich string is all indented
and ignore the first tabs, but it is tricky.
I don't fully understand how that works :S

I mean I can indent the first lines but not the ones that have the string
I'm comparing in the assertion, they will introduce spaces.

On Tue, Jul 12, 2016 at 8:22 PM, Nico Passerini <notifications@github.com>
wrote:

> In
> org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/natives/ExceptionPropagationInNativeTestCase.xtend
> <https://github.com/uqbar-project/wollok/pull/811#discussion_r70540536>:
>
> > +	const repeater = new Repeater()
> > +
> > +	const closure = { throw new Exception("SoundGarden") }
> > +	try {
> > +		repeater.ntimes(1, closure)
> > +	}
> > +	catch e {
> > +		assert.equals(
> > +"wollok.lang.Exception: SoundGarden
> > +	at  [natives.wpgm]
> > +	at natives.Repeater.ntimes(n,closure) [natives.wpgm]
> > +	at  [natives.wpgm]
> > +",e.getStackTraceAsString())
> > +	}
> > +}
> > +		''']
>
> Why can't we take advantage of xtend ''' and use better indentation?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/uqbar-project/wollok/pull/811/files/2d7cd933682ec394926a49fdff0313058d3e4998#r70540536>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEORWKA3wPUIF-W4Ju_8v9gGIJiaaegZks5qVCGYgaJpZM4JK6x5>
> .
>